1.illumination_correction/scripts/0.create_loaddata_csvs.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Create LoadData CSVs to use for illumination correction
+# 
+# In this notebook, we create a LoadData CSV that contains paths to each channel per image set for CellProfiler to process. 
+# We can use this LoadData CSV to run illumination correction (IC) pipeline that saves IC functions in `npy` file format and extract image quality metrics.
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+import pandas as pd
+import re
+
+import sys
+
+sys.path.append("../utils")
+import loaddata_utils as ld_utils
+
+
+# ## Set paths
+
+# In[2]:
+
+
+# Paths for parameters to make loaddata csv
+index_directory = pathlib.Path("/media/18tbdrive/ALSF_pilot_data/SN0313537/")
+config_path = pathlib.Path("./config.yml").absolute()
+output_csv_dir = pathlib.Path("./loaddata_csvs")
+output_csv_dir.mkdir(parents=True, exist_ok=True)
+
+# Find all 'Images' folders within the directory
+images_folders = list(index_directory.rglob('Images'))
+
+
+# ## Create LoadData CSVs for all data
+
+# In[3]:
+
+
+# Loop through each folder and create a LoadData CSV
+for folder in images_folders:
+    # Get the first folder directly under the index_directory
+    relative_path = folder.relative_to(index_directory)
+    first_folder = relative_path.parts[0]  # First-level folder
+
+    # Generate the plate name based on folder structure
+    if first_folder.startswith('BR00'):
+        plate_name = first_folder.split('_')[0]  # Take the first part
+
+    elif first_folder.startswith('2024'):
+        second_folder = relative_path.parts[1]  # Second-level folder
+        part1 = '_'.join(first_folder.split('_')[-2:])  # Last two parts of first folder
+        part2 = second_folder.split('_')[0]  # First part of second folder
+        plate_name = f"{part1}_{part2}"  # Combine them
+
+    else:
+        print(f"Unexpected folder pattern: {folder}")
+        continue  # Skip if not matching patterns
```

Comment:
Consider adding loggin? Up to you.

Reply:
It is a good thought! I think for this notebook, it will probably be a temporary fix for this preliminary data so I will hold off on adding logging here unless it get more complicated.